Description: disable sixel functionality
 Support for sixel requires packaging the icy_sixel crate, since it's
 not yet packaged in Debian. This crate is currently only used as a
 dependency of doxx (ITP: #1113781), which does not require sixel. As
 no other package depends on this feature, sixel support is disabled
 for now but may be re-enabled in the future.
Author: Nadzeya Hutsko <nadzya.info@gmail.com>
Forwarded: not-needed
Last-Update: 2025-09-05
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,9 +65,6 @@
 [dependencies.base64]
 version = "^0.22.1"
 
-[dependencies.icy_sixel]
-version = "^0.1.1"
-
 [dependencies.image]
 version = "^0.25.5"
 features = ["jpeg"]
--- a/src/picker.rs
+++ b/src/picker.rs
@@ -12,7 +12,6 @@
         halfblocks::Halfblocks,
         iterm2::Iterm2,
         kitty::{Kitty, StatefulKitty},
-        sixel::Sixel,
         Protocol, StatefulProtocol, StatefulProtocolType,
     },
     FontSize, ImageSource, Resize, Result,
@@ -30,8 +29,6 @@
 pub enum Capability {
     /// Reports supporting kitty graphics protocol.
     Kitty,
-    /// Reports supporting sixel graphics protocol.
-    Sixel,
     /// Reports supporting rectangular ops.
     RectangularOps,
     /// Reports font size in pixels.
@@ -60,7 +57,6 @@
 )]
 pub enum ProtocolType {
     Halfblocks,
-    Sixel,
     Kitty,
     Iterm2,
 }
@@ -68,8 +64,7 @@
 impl ProtocolType {
     pub fn next(&self) -> ProtocolType {
         match self {
-            ProtocolType::Halfblocks => ProtocolType::Sixel,
-            ProtocolType::Sixel => ProtocolType::Kitty,
+            ProtocolType::Halfblocks => ProtocolType::Kitty,
             ProtocolType::Kitty => ProtocolType::Iterm2,
             ProtocolType::Iterm2 => ProtocolType::Halfblocks,
         }
@@ -220,7 +215,6 @@
 
         match self.protocol_type {
             ProtocolType::Halfblocks => Ok(Protocol::Halfblocks(Halfblocks::new(image, area)?)),
-            ProtocolType::Sixel => Ok(Protocol::Sixel(Sixel::new(image, area, self.is_tmux)?)),
             ProtocolType::Kitty => Ok(Protocol::Kitty(Kitty::new(
                 image,
                 area,
@@ -236,10 +230,6 @@
         let source = ImageSource::new(image, self.font_size, self.background_color);
         let protocol_type = match self.protocol_type {
             ProtocolType::Halfblocks => StatefulProtocolType::Halfblocks(Halfblocks::default()),
-            ProtocolType::Sixel => StatefulProtocolType::Sixel(Sixel {
-                is_tmux: self.is_tmux,
-                ..Sixel::default()
-            }),
             ProtocolType::Kitty => {
                 StatefulProtocolType::Kitty(StatefulKitty::new(random(), self.is_tmux))
             }
@@ -457,13 +447,6 @@
                 proto = Some(ProtocolType::Kitty);
                 Some(Capability::Kitty)
             }
-            Response::Sixel => {
-                if proto.is_none() {
-                    // Only if kitty is not supported.
-                    proto = Some(ProtocolType::Sixel);
-                }
-                Some(Capability::Sixel)
-            }
             Response::RectangularOps => Some(Capability::RectangularOps),
             Response::CellSize(cell_size) => {
                 if let Some((w, h)) = cell_size {
@@ -547,8 +530,6 @@
     fn test_cycle_protocol() {
         let mut proto = ProtocolType::Halfblocks;
         proto = proto.next();
-        assert_eq!(proto, ProtocolType::Sixel);
-        proto = proto.next();
         assert_eq!(proto, ProtocolType::Kitty);
         proto = proto.next();
         assert_eq!(proto, ProtocolType::Iterm2);
--- a/src/protocol.rs
+++ b/src/protocol.rs
@@ -12,7 +12,6 @@
     halfblocks::Halfblocks,
     iterm2::Iterm2,
     kitty::{Kitty, StatefulKitty},
-    sixel::Sixel,
 };
 use crate::{FontSize, ResizeEncodeRender, Result};
 
@@ -21,7 +20,6 @@
 pub mod halfblocks;
 pub mod iterm2;
 pub mod kitty;
-pub mod sixel;
 
 trait ProtocolTrait: Send + Sync {
     /// Render the currently resized and encoded data to the buffer.
@@ -44,7 +42,6 @@
 #[derive(Clone)]
 pub enum Protocol {
     Halfblocks(Halfblocks),
-    Sixel(Sixel),
     Kitty(Kitty),
     ITerm2(Iterm2),
 }
@@ -53,7 +50,6 @@
     pub(crate) fn render(&self, area: Rect, buf: &mut Buffer) {
         let inner: &dyn ProtocolTrait = match self {
             Self::Halfblocks(halfblocks) => halfblocks,
-            Self::Sixel(sixel) => sixel,
             Self::Kitty(kitty) => kitty,
             Self::ITerm2(iterm2) => iterm2,
         };
@@ -62,7 +58,6 @@
     pub fn area(&self) -> Rect {
         let inner: &dyn ProtocolTrait = match self {
             Self::Halfblocks(halfblocks) => halfblocks,
-            Self::Sixel(sixel) => sixel,
             Self::Kitty(kitty) => kitty,
             Self::ITerm2(iterm2) => iterm2,
         };
@@ -85,7 +80,6 @@
 #[derive(Clone)]
 pub enum StatefulProtocolType {
     Halfblocks(Halfblocks),
-    Sixel(Sixel),
     Kitty(StatefulKitty),
     ITerm2(Iterm2),
 }
@@ -94,7 +88,6 @@
     fn inner_trait(&self) -> &dyn StatefulProtocolTrait {
         match self {
             Self::Halfblocks(halfblocks) => halfblocks,
-            Self::Sixel(sixel) => sixel,
             Self::Kitty(kitty) => kitty,
             Self::ITerm2(iterm2) => iterm2,
         }
@@ -102,7 +95,6 @@
     fn inner_trait_mut(&mut self) -> &mut dyn StatefulProtocolTrait {
         match self {
             Self::Halfblocks(halfblocks) => halfblocks,
-            Self::Sixel(sixel) => sixel,
             Self::Kitty(kitty) => kitty,
             Self::ITerm2(iterm2) => iterm2,
         }
--- a/src/picker/cap_parser.rs
+++ b/src/picker/cap_parser.rs
@@ -16,7 +16,6 @@
 #[derive(Debug, PartialEq, Clone)]
 pub enum Response {
     Kitty,
-    Sixel,
     RectangularOps,
     CellSize(Option<(u16, u16)>),
     CursorPositionReport(u16, u16),
@@ -126,7 +125,6 @@
                         let inner: Vec<&str> = (self.data[2..]).split(';').collect();
                         for cap in inner {
                             match cap {
-                                "4" => caps.push(Response::Sixel),
                                 "28" => caps.push(Response::RectangularOps),
                                 _ => {}
                             }
@@ -220,7 +218,6 @@
             caps,
             vec![
                 Response::Kitty,
-                Response::Sixel,
                 Response::CellSize(Some((14, 7))),
                 Response::CursorPositionReport(6, 6),
                 Response::CursorPositionReport(7, 7),
